core/services/remote_access_services.py

```diff
@@ -0,0 +1,128 @@
+
+
+def _get_url():
+    if vmconf.DEV_MODE:
+        return vmconf.DEFAULT_COMMUNICATION_URL
+
+    return vmconf.SERVER_COMMUNICATION_URL
+
+
+def _get_port():
+    if vmconf.DEV_MODE:
+        return vmconf.DEFAULT_COMMUNICATION_PORT
+
+    return vmconf.SERVER_COMMUNICATION_PORT
+
+
+def _get_vm_id():
+    if vmconf.DEV_MODE:
+        return vmconf.DEFAULT_VM_ID
+
+    # Get VMID dynamically from metadata.
+    return metadata_services.get_metadata_param(
+        vmconf.METADATA_VM_ID_PARAM_NAME)
+
+
+def _get_shared_secret():
+    if vmconf.DEV_MODE:
+        return vmconf.DEFAULT_VM_SHARED_SECRET
+
+    # Get shared secret dynamically from metadata.
+    return metadata_services.get_metadata_param(
+        vmconf.METADATA_SHARED_SECRET_PARAM_NAME)
+
+
+def generate_signature(data):
+    """Generates digital signature for given data.
+
+    Args:
+        data: dict. A dictionary data to be transferred over network.`
+    """
+    msg = json.dumps(data)
+    key = _get_shared_secret()
+
+    # Generate signature and return it.
+    return hmac.new(key, msg, digestmod=hashlib.sha256).hexdigest()
+
+
+def fetch_next_job_request():
+    """Returns the next job request to be processed.
+
+    Returns:
+        dict. A dict retrieved remotely from database containing
+        job request data.
+    """
+
+    request_url = "%s:%s/%s" % (
+        _get_url(), _get_port(), vmconf.FETCH_NEXT_JOB_REQUEST_HANDLER)
+
+    payload = {
+        'vm_id': _get_vm_id()
+    }
+    signature = generate_signature(payload)
+    payload['signature'] = signature
+    response = requests.get(request_url, params=payload)
+    return response.json()
+
+
+def store_trained_classifier_model(classifier_data):
+    """Stores the result of processed job request.
+
+    Args:
+        classifier_data: dict. A dictionary containing result of training
+            of classifier.
+
+    Returns:
+        response: response object containing the server's response.
+
+    Raises:
+        Exception: classifier_data is not of dict type.
+        Exception: classifier_data does not contain 'job_request_id' key.
+        Exception: classifier_data does not contain 'training_result' key.
+    """
+
+    # Make sure that classifier_data is in proper foramt.
+    if not isinstance(classifier_data, dict):
+        raise Exception('classifier_data must be in dict format.')
+
+    if 'job_request_id' not in classifier_data.keys():
+        raise Exception('classifier_data must contain \'job_request_id\'.')
+
+    if 'training_result' not in classifier_data.keys():
+        raise Exception('classifier_data must contain \'training_result\'.')
+
+    payload = classifier_data
+    payload['vm_id'] = _get_vm_id()
+    signature = generate_signature(payload)
+    payload['signature'] = signature
+    request_url = "%s:%s/%s" % (
+        _get_url(), _get_port(), vmconf.STORE_TRAINED_CLASSIFIER_MODEL_HANDLER)
+    response = requests.post(request_url, json=payload)
```

Comment:
Yes it only contains job result. I have already mentioned the behavior of the controller in doc [here](https://docs.google.com/document/d/1kACmivEvZVlTYBdqxy-mmdTi3sg_NbkZdS10T-ya-pg/edit#heading=h.yu0j0s1lpm2p).